src/CMakeLists.txt

```diff
@@ -627,39 +648,109 @@ else()
-    add_doom_tiny("" render_newhope)
     add_library(tiny_settings INTERFACE)
     target_compile_definitions(tiny_settings INTERFACE
-            WHD_SUPER_TINY=1
-            DEMO1_ONLY=1
-            NO_USE_FINALE_CAST=1
-            NO_USE_FINALE_BUNNY=1
-    )
-    target_compile_definitions(doom_tiny PRIVATE
-            TINY_WAD_ADDR=0x10040000
+        WHD_SUPER_TINY=1
+        DEMO1_ONLY=1
+        NO_USE_FINALE_CAST=1
+        NO_USE_FINALE_BUNNY=1
     )
-    target_link_libraries(doom_tiny PRIVATE tiny_settings)
-    add_doom_tiny(_nost render_newhope)
+endif()
 
-    target_compile_definitions(doom_tiny_nost PRIVATE
-            # NO_USE_FINALE_CAST=1 # note this is only used by doom II but doesnt take that much more space
-            TINY_WAD_ADDR=0x10048000
+function(add_doom_tiny_screen SCREEN)
+    if (PICO_SDK)
+        # add_doom_tiny(_${SCREEN} ${SCREEN})
+
+        # target_compile_definitions(doom_tiny_${SCREEN} PRIVATE
+        #         TINY_WAD_ADDR=0x10040000
+        # )
+        # target_link_libraries(doom_tiny_${SCREEN} PRIVATE tiny_settings)
+
+        # disabled because I don't need it, probably still works...
+        # add_doom_tiny(_nost_${SCREEN} ${SCREEN})
+
+        # target_compile_definitions(doom_tiny_nost_${SCREEN} PRIVATE
+        #         # NO_USE_FINALE_CAST=1 # note this is only used by doom II but doesnt take that much more space
+        #         TINY_WAD_ADDR=0x10048000
+        #         )
+        if (PICO_ON_DEVICE)
+            suppress_tinyusb_warnings()
+            add_doom_tiny(_usb_${SCREEN} ${SCREEN})
+            target_link_libraries(doom_tiny_usb_${SCREEN} PRIVATE tiny_settings pico_cd)
+            target_compile_definitions(doom_tiny_usb_${SCREEN} PRIVATE 
+                    USB_SUPPORT=1
+                    TINY_WAD_ADDR=0x10042000 # ugh tinyusb is >8K currently!!
             )
-    if (PICO_ON_DEVICE)
-        suppress_tinyusb_warnings()
-        add_doom_tiny(_usb render_newhope)
-        target_link_libraries(doom_tiny_usb PRIVATE tiny_settings pico_cd)
-        target_compile_definitions(doom_tiny_usb PRIVATE
-                USB_SUPPORT=1
-                TINY_WAD_ADDR=0x10042000 # ugh tinyusb is >8K currently!!
-        )
-        add_doom_tiny(_nost_usb render_newhope)
-        target_link_libraries(doom_tiny_nost_usb PRIVATE pico_cd)
-        target_compile_definitions(doom_tiny_nost_usb PRIVATE
-                USB_SUPPORT=1
-                TINY_WAD_ADDR=0x10048000
-                )
+            # disabled for same reason as above
+            # add_doom_tiny(_nost_usb_${SCREEN} ${SCREEN})
+            # target_link_libraries(doom_tiny_nost_usb_${SCREEN} PRIVATE pico_cd)
+            # target_compile_definitions(doom_tiny_nost_usb_${SCREEN} PRIVATE
+            #         USB_SUPPORT=1
+            #         TINY_WAD_ADDR=0x10048000
+            #         )
+        endif()
     endif()
-endif()
+endfunction()
 
 
+
+
+add_doom_tiny_screen("SSD1306_70_40")
```

Comment:
this is an example declaration for a screen. If you wanted to make a new screen, you'd have to make one of these.

You'll notice the target includes the resolution for the screen - I'm not sure how I feel about that. The downsampling factor definitely needs to be set, which implicitly determines the resolution of the screen, but I might further break out a system for targeting different screen resolutions for the same controller